File: android-tech-test-java/app/src/main/java/com/bridge/androidtechnicaltest/db/PupilDao.java
package com.bridge.androidtechnicaltest.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Transaction;

import java.util.List;

import io.reactivex.Single;

@Dao
public interface PupilDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<Pupil> pupilList);

    @Query("SELECT * FROM Pupils ORDER BY name ASC")
    Single<List<Pupil>> getPupils();

    @Query("DELETE FROM Pupils")
    void clear();
}

File: android-tech-test-kotlin/app/src/main/java/com/bridge/androidtechnicaltest/db/PupilDao.kt
package com.bridge.androidtechnicaltest.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Single

@Dao
interface PupilDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(pupilList: List<Pupil>)

    @Query("SELECT * FROM Pupils ORDER BY name ASC")
    fun getPupils(): Single<List<Pupil>>

    @Query("SELECT * FROM Pupils ORDER BY name ASC")
    suspend fun getAllPupils(): List<Pupil>

    @Query("DELETE FROM Pupils")
    fun clear()
}